test/docs/sonic-dash-test-HLD.md

```diff
@@ -0,0 +1,212 @@
+![SONiC Architecture](../images/dash-thrift-sai-api.svg)
+
+## Non-standardized DUT Configuration Methodology
+Non-standard, vendor-specific APIs/utilities etc. are not officially recognized. However, for practical reasons, it should be possible to utilize non-standard DUT setup techniques in conjunction with the standardized community test cases and methodology. Towards this end, test cases and scripts *should* be structured such that it is relatively easy to substitute non-standard device setup steps with a vendor-supplied version. For example, a Pytest fixture setup class, or an imported wrapper library with subclassed methods could be instantiated to hide such details from the overall test framework.
+
+>**NOTE**: It is incumbent on prospective suppliers of such non-standard devices/management software, to implement and integrate such accommodations in a manner which does not detract from the overall elegance, efficiency and maintainability of this repository. Code contributions to make these changes must not impose undue burden on the community at large.
+
+# Test Automation Framework
+Test Automation shall utilize the [PyTest](https://docs.pytest.org/en/6.2.x/) Framework.
+
+**TODO** More details, replace "shall" language with actual guidelines.
+
+# Test Objectives
+## Conformance Testing
+Conformance tests verify that the DUT performs all functions correctly, handles negative tests properly and obeys specified APIs. The list below summarizes the conformance tests.
+
+**TODO** Define
+
+## Performance Testing
+Performance tests measure the DUT's capacity, speed and scale limits under various conditions. The list below summarizes the performance tests.
+
+**TODO** Define
+
+# Standardized Test Cases
+## Test Case Organization
+Test cases shall be organized in an appropriate manner. File directory structures and conventions shall be established.
+
+**TODO** More details
+
+## Test Case Marking
+Tests shall be tagged with appropriate markers to allow selecting or deselecting categories of tests to run particular suites.
+
+**TODO** More details, replace "shall" language with actual guidelines.
+
+## Vendor/Platform Agnostic Traffic Generators
+Where feasible, test should be designed against an abstract, vendor-neutral, platform-neutral Traffic Generator data model and API, to avoid lock-in. The [OpenTrafficGenerator](https://github.com/open-traffic-generator) project provides suitable data models and the derived [snappi libraries](https://github.com/open-traffic-generator/snappi) simplify writing client interfaces. Free (SW-only) and commercial (SW and HW) traffic generators adhering to the OTG specifications are available. These comprise the recommended solutions. 
+
+When traffic test stimuli demand non-agnostic traffic-generator solutions, suitable  commercial solutions shall be utilized as required. Test automation software *should* implement generic wrappers in the main code to reduce vendor lock-in.
```

Comment:
for the testbed setup for example (lag, bgp session, etc. on), let's try to use the sonic-mgmt repo to set it up, instead of relying on the traffic generator.

Reply:
Guohan, good point, I did not differentiate between dataplane and control plane. We can discuss a bit more and I will articulate this in the spec better. I agree the Tgen was meant for dataplane and the control plane can leverage existing testbed infrastructure.

Reply:
I added some TODOs in the HLD to ensure we handle this as we move forward.